doc/source/serve/tutorials/gradio-integration.md

```diff
@@ -0,0 +1,270 @@
+```bash
+$ pip install transformers
+```
+
+
+Then, we construct the Gradio app `io`:
+```python
+from transformers import pipeline
+
+summarizer = pipeline("summarization", model="t5-small")
+def model(text):
+    summary_list = summarizer(text)
+    summary = summary_list[0]["summary_text"]
+    return summary
+
+example = (
+    "HOUSTON -- Men have landed and walked on the moon. "
+    "Two Americans, astronauts of Apollo 11, steered their fragile "
+    "four-legged lunar module safely and smoothly to the historic landing "
+    "yesterday at 4:17:40 P.M., Eastern daylight time. Neil A. Armstrong, the "
+    "38-year-old commander, radioed to earth and the mission control room "
+    'here: "Houston, Tranquility Base here. The Eagle has landed." The '
+    "first men to reach the moon -- Armstrong and his co-pilot, Col. Edwin E. "
+    "Aldrin Jr. of the Air Force -- brought their ship to rest on a level, "
+    "rock-strewn plain near the southwestern shore of the arid Sea of "
+    "Tranquility. About six and a half hours later, Armstrong opened the "
+    "landing craft's hatch, stepped slowly down the ladder and declared as "
+    "he planted the first human footprint on the lunar crust: \"That's one "
+    'small step for man, one giant leap for mankind." His first step on the '
+    "moon came at 10:56:20 P.M., as a television camera outside the craft "
+    "transmitted his every move to an awed and excited audience of hundreds "
+    "of millions of people on earth."
+)
+
+io = gr.Interface(
+    fn=model,
+    inputs=[gr.inputs.Textbox(
+        default=example, label="Input prompt"
+    )],
+    outputs=[gr.outputs.Textbox(label="Model output")]
+)
+```
+
+
+### Understanding `GradioIngress`
+In order to deploy your Gradio App onto a Ray Cluster, you need to wrap your Gradio App in a serve deployment. `GradioIngress` acts as that wrapper. It serves your Gradio App on a Ray cluster.
+```python
+class GradioIngress:
+    def __init__(self, io: gr.Blocks):
+        self.app = gr.routes.App.create_app(io)
+
+    async def __call__(self, request: starlette.requests.Request):
+        sender = ASGIHTTPSender()
+        await self.app(request.scope, receive=request.receive, send=sender)
+        return sender.build_asgi_response()
+```
+
+### Define your Gradio Server
+First, we need to define a deployable Gradio Server built on top of `GradioIngress`. As this example is demonstrating its most basic use as a wrapper, we just need to decorate `GradioIngress` as a serve deployment.
+
+```python
+import gradio as gr
+from ray.serve.gradio import GradioIngress
+
+GradioServer = serve.deployment(GradioIngress)
+```
+
+### Deploy your Gradio Server
+First, Ray Serve needs to run on top of a Ray cluster, so we connect to a local one.
+
+```python
+ray.init(address="auto")
+```
+
+Next, we start the Ray Serve runtime:
+
+```python
+serve.start(detached=True)
```

Comment:
You don't need to do either of these if you use `serve run` CLI. I'd suggest doing that for this example and link out for instructions on how to use it on a cluster. The CLI command would be `serve run file:gradio_app`